### ISSUE TITLE
Remove pi_interval

### ISSUE THREAD

**Issue by dpsanders** [JuliaIntervals/IntervalArithmetic.jl#285]
`pi_interval(T)` is the same as `Interval{T}(pi)`.

cc @yashrajgupta

**Comment by dpsanders** [comment]
Also we can use `oftype(a, pi)` if `a` is an `Interval`.

**Comment by yashrajgupta** [comment]
Seems like something is different, 
julia> pi_interval(Float64)
Interval(3.141592653589793, 3.1415926535897936)

julia> Interval{Float64}(pi)
Interval(3.141592653589793, 3.141592653589793)

**Comment by dpsanders** [comment]
Ah, oops. Could you fix this please?

**Comment by dpsanders** [comment]
(Please quote code inside triple backticks with `jl` after the first set.)

\`\`\`jl
(code)
\`\`\`

**Comment by yashrajgupta** [comment]
I think pi_interval is a function which we have defined but Interval{T}(pi) is a type conversion of a rational number pi in to an Interval of type T. So what do we want at the end? Is it that pi_interval should be interval of one absolute value of pi or should it an interval containing pi and nearby values.

**Comment by dpsanders** [comment]
Of course we want it to be the smallest interval containing `pi`, which is obtained currently from
```
julia> convert(Interval, pi)
Interval(3.141592653589793, 3.1415926535897936)
```
(although who knows how efficient that is).

[Please quote code inline as \`(code)\`.]

**Comment by dpsanders** [comment]
The idea is always that interval arithmetic should give results that are guaranteed to contain the true results. In this case `pi` is an irrational (not rational, as you wrote) number, which is not representable exactly in floating point. So any interval calculations with `pi` must use an interval containing `pi`, preferably the smallest possible one.

**Comment by yashrajgupta** [comment]
So , probably we should remove 'pi_interval' from everywhere and use 'Interval{T}(pi)' instead. Or we can change the definition of 'pi_interval' as 'pi_interval(T) = Interval{T}(pi)' .

**Comment by dpsanders** [comment]
Hmmm. It seems that `convert` here is slow. Maybe we should retain `pi_interval(T)` and just calculate the interval once and for all statically.

**Comment by yashrajgupta** [comment]
Sorry about the typo. Yes we should retain '(pi_interval)' function until we find a smaller interval for pi moreover I think '(Interval{T}(pi))' does not contain the pi value since it is just a single value interval.

**Comment by dpsanders** [comment]
I am saying that this is the smallest interval for `Float64`:
```
julia> x = convert(Interval{Float64}, pi)
Interval(3.141592653589793, 3.1415926535897936)
```
You can test this:
```jl
julia> nextfloat(x.lo) == x.hi
true
```

**Comment by dpsanders** [comment]
Ah, I see that `pi_interval` already does what I was suggesting -- defines it once and for all.

**Comment by dpsanders** [comment]
But we should define `Interval{Float64}(pi)` and `Interval(pi)` to do the same thing.

**Comment by lbenet** [comment]
I think `pi_interval` here is ok; the version for `Float64` uses `atomic(Interval{Float64}, pi)` which is more efficient than convert. Perhaps this should be better explained in the documentation.

**Comment by Kolaru** [comment]
In #271 I plan to get rid of `pi_interval` by defining (for each flavor but here is the version for `Interval`)

```jl
Interval{T}(::Irrational{:π}) where T = atomic(Interval{T}, π)
Interval(::Irrational{:π}) = Interval{Float64}(π)
```

If further optimisation is required, the methods are specific enough to have them there.

**Comment by lbenet** [comment]
Closed by #338
